core/src/test/java/io/methvin/watcher/DirectoryWatcherOnIdleTest.java

```diff
@@ -0,0 +1,214 @@
+    AtomicLong updated = new AtomicLong();
+    AtomicInteger counter = new AtomicInteger(-1);
+    AtomicInteger onIdleCalled = new AtomicInteger(0);
+
+    Consumer<Integer> consumer =
+        value -> {
+          updated.set(System.currentTimeMillis() - started.get());
+          counter.set(value);
+          onIdleCalled.incrementAndGet();
+        };
+
+    watcher =
+        DirectoryWatcher.builder()
+            .paths(Arrays.asList(d1))
+            .listener(new CompositeListener(1000, consumer))
+            .fileHashing(true)
+            .build();
+
+    started.set(System.currentTimeMillis());
+    watcher.watchAsync();
+
+    Files.createTempFile(d1, "parent1-", ".dat");
+    doDelay(800);
+    Files.createTempFile(d1, "parent2-", ".dat");
+    Files.createTempFile(d1, "parent3-", ".dat");
+    doDelay(1200);
+    Files.createTempFile(d1, "parent4-", ".dat");
+    doDelay(100);
+    Files.createTempFile(d1, "parent5-", ".dat");
+    doDelay(2000);
+    Files.createTempFile(d1, "parent6-", ".dat");
+    doDelay(800);
+    Files.createTempFile(d1, "parent7-", ".dat");
+    doDelay(1100);
+
+    assertEquals(3, onIdleCalled.get());
+    assertEquals(isMac() ? 15 : 7, counter.get());
+    watcher.close();
+  }
+
+  private boolean isMac() {
+    return System.getProperty("os.name").toLowerCase().contains("mac");
+  }
+
+  @Test
+  public void testOnIdleChangeSet() throws IOException {
+    Assume.assumeTrue(!isMac());
+
+    Path d1 = this.tmpDir.resolve("idle3");
+    Files.createDirectory(d1);
+
+    AtomicLong started = new AtomicLong();
+    AtomicLong updated = new AtomicLong();
+    AtomicInteger counter = new AtomicInteger(-1);
+    AtomicInteger onIdleCalled = new AtomicInteger(0);
+
+    CompositeListener composite =
+        new CompositeListener(
+            100,
+            value -> {
+              updated.set(System.currentTimeMillis() - started.get());
+              counter.set(value);
+              onIdleCalled.incrementAndGet();
+            });
+
+    watcher =
+        DirectoryWatcher.builder()
+            .paths(Arrays.asList(d1))
+            .listener(composite)
+            .fileHashing(true)
+            .build();
+
+    started.set(System.currentTimeMillis());
+    watcher.watchAsync();
+
+    Files.createTempFile(d1, "parent1-", ".dat");
+    Files.createTempFile(d1, "parent2-", ".dat");
+    Files.createTempFile(d1, "parent3-", ".dat");
+    Files.createTempFile(d1, "parent4-", ".dat");
+    Files.createTempFile(d1, "parent5-", ".dat");
+    Files.createTempFile(d1, "parent6-", ".dat");
+    Files.createTempFile(d1, "parent7-", ".dat");
+    doDelay(200);
+
+    assertEquals(1, onIdleCalled.get());
+
+    Map<Path, ChangeSet> changeSet = composite.getChangeSet();
+    assertTrue(!changeSet.isEmpty());
+    assertEquals(7, changeSet.get(d1).created().size());
+
+    watcher.close();
+  }
+
+  private class CompositeListener implements DirectoryChangeListener {
```

Comment:
I thought we were talking about adding a helper that creates a listener out of a list of listeners? Seems like a natural addition to the API.